llvm/test/Transforms/CodeGenPrepare/revert-constant-ptr-propagation-on-calls.ll

```diff
@@ -0,0 +1,123 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py
+; RUN: opt -S -passes='require<profile-summary>,function(codegenprepare)' -mtriple=aarch64-none-linux-gnu < %s | FileCheck %s
+
+%struct.S = type { i8 }
+%struct.X = type { i32 }
+
+@g_getS = internal global %struct.S zeroinitializer, align 1
+@g_getX = internal global %struct.X zeroinitializer, align 1
+@guard = internal global i64 0, align 8
+
+declare ptr @getS_dec()
+
+define nonnull align 1 dereferenceable(1) ptr @getS() personality ptr @__gxx_personality_v0 {
```

Comment:
We should also have coverage for the logic that detects the return value, e.g. a function with two different return values as a negative test.

Reply:
Should be taken into account in test function `caller_4`.